Sources/Extensions/UIView+PinLayout.swift

```diff
@@ -95,4 +102,42 @@ extension UIView: Layoutable, SizeCalculable {
     }
 }
 
+extension UIView: AutoSizeCalculable {
+    private struct pinlayoutAssociatedKeys {
+        static var pinlayoutAutoSizingRect = UnsafeMutablePointer<Int8>.allocate(capacity: 1)
+        static var pinlayoutAutoSizingRectWithMargins = UnsafeMutablePointer<Int8>.allocate(capacity: 1)
+    }
+
+    public var autoSizingRect: CGRect? {
+        get {
+            return objc_getAssociatedObject(self, &pinlayoutAssociatedKeys.pinlayoutAutoSizingRect) as? CGRect
+        }
+        set {
+            objc_setAssociatedObject(self, &pinlayoutAssociatedKeys.pinlayoutAutoSizingRect, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+
+    public var autoSizingRectWithMargins: CGRect? {
+        get {
+            return objc_getAssociatedObject(self, &pinlayoutAssociatedKeys.pinlayoutAutoSizingRectWithMargins) as? CGRect
+        }
+        set {
+            objc_setAssociatedObject(self, &pinlayoutAssociatedKeys.pinlayoutAutoSizingRectWithMargins, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+
+    public func autoSizeThatFits(_ size: CGSize, layoutClosure: () -> Void) -> CGSize {
```

Comment:
I really would like to extract that function out of the UIView conformance extension but the only thing that prevent me to do so is the call to `let adjustedRect = Coordinates<View>.adjustRectToDisplayScale(rect)`. I think it would make sense to add a `displayScale` property on the `Layoutable` protocol and perform that transformation before calling `setRect`. We really want the resulting rect to be the same in both layout and auto sizing.

Reply:
I have perform the changes to do this but I believe it would make more sense in a separate PR as it also fix an issue in the current production code.